core/src/main/java/org/apache/brooklyn/core/workflow/WorkflowStepResolution.java

```diff
@@ -90,7 +91,18 @@ static WorkflowStepDefinition resolveStep(ManagementContext mgmt, Object def) {
                 if (s == null) s = defM.remove("shorthand");
                 if (s == null) s = defM.remove("s");
                 if (s == null) {
-                    throw new IllegalArgumentException("Step definition must indicate a `type` or a `step` / `shorthand` / `s` ("+def+")");
+                    if (defM.size()==1) {
+                        // assume the colon caused it accidentally to be a map
+                        s = Iterables.getOnlyElement(defM.keySet());
+                        if (s instanceof String && ((String)s).contains(" ")) {
```

Comment:
@ahgittin Probably could replace this `if` with an elvis operator.

Reply:
in this (rare!) instance i think the longer-winded syntax is worth it for readability